rocAL/rocAL_pybind/amd/rocal/pipeline.py

```diff
@@ -241,28 +215,81 @@ def copyImage(self, array):
         b.raliCopyToOutput(
             self._handle, np.ascontiguousarray(out, dtype=array.dtype))
 
-
-
     def copyToTensor(self, array,  multiplier, offset, reverse_channels, tensor_format, tensor_dtype):
 
         b.raliCopyToOutputTensor(self._handle, ctypes.c_void_p(array.data_ptr()), tensor_format, tensor_dtype,
                                     multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0))
 
+    def copyToTensorNHWC(self, array,  multiplier, offset, reverse_channels, tensor_dtype):
+        out = np.frombuffer(array, dtype=array.dtype)
+        if tensor_dtype == types.FLOAT:
+            b.raliCopyToOutputTensor32(self._handle, np.ascontiguousarray(out, dtype=array.dtype), types.NHWC,
+                                       multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0))
+        elif tensor_dtype == types.FLOAT16:
+            b.raliCopyToOutputTensor16(self._handle, np.ascontiguousarray(out, dtype=array.dtype), types.NHWC,
+                                       multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0))
+
+    def copyToTensorNCHW(self, array,  multiplier, offset, reverse_channels, tensor_dtype):
+        out = np.frombuffer(array, dtype=array.dtype)
+        if tensor_dtype == types.FLOAT:
+            b.raliCopyToOutputTensor32(self._handle, np.ascontiguousarray(out, dtype=array.dtype), types.NCHW,
+                                       multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0))
+        elif tensor_dtype == types.FLOAT16:
+            b.raliCopyToOutputTensor16(self._handle, np.ascontiguousarray(out, dtype=array.dtype), types.NCHW,
+                                       multiplier[0], multiplier[1], multiplier[2], offset[0], offset[1], offset[2], (1 if reverse_channels else 0))
+
     def encode(self, bboxes_in, labels_in):
         bboxes_tensor = torch.tensor(bboxes_in).float()
         labels_tensor=  torch.tensor(labels_in).long()
         return self._encode_tensor.prev.rali_c_func_call(self._handle, bboxes_tensor , labels_tensor )
 
     def GetOneHotEncodedLabels(self, array):
-        return b.getOneHotEncodedLabels(self._handle, array, self._numOfClasses)
+        return b.getOneHotEncodedLabels(self._handle, ctypes.c_void_p(array.data_ptr()), self._numOfClasses)
+
+    def set_outputs(self, *output_list):
+        self._output_list_length = len(output_list)
+        b.setOutputImages(self._handle,len(output_list),output_list)
+
+    def __enter__(self):
+        Pipeline._current_pipeline = self
+        return self
 
+    def __exit__(self, exception_type, exception_value, traceback):
+        pass
+
+    def set_seed(self,seed=0):
```

Comment:
this seems to be incorrect. set_seed() is not global. It should set_seed() for each pipeline which means that it needs the handle